tests/unit/test_execution_environment_utils.py

```diff
@@ -0,0 +1,292 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import logging
+import os
+import pytest
+
+from galaxy_importer.utils import execution_environment as ee_utils
+
+
+logger = logging.getLogger(__name__)
+
+
+EE_YAML = """---
+version: 1
+dependencies:
+  galaxy: requirements.yml
+  python: requirements.txt
+  system: bindep.txt
+
+additional_build_steps:
+  prepend: |
+    RUN whoami
+    RUN cat /etc/os-release
+  append:
+    - RUN echo This is a post-install command!
+    - RUN ls -la /etc
+"""
+
+
+EE_DIR_YAML = """---
+version: 1
+dependencies:
+  galaxy: meta/requirements.yml
+  python: meta/requirements.txt
+  system: meta/bindep.txt
+
+additional_build_steps:
+  prepend: |
+    RUN whoami
+    RUN cat /etc/os-release
+  append:
+    - RUN echo This is a post-install command!
+    - RUN ls -la /etc
+"""
+
+REQUIREMENTS_YML = """
+---
+collections:
+  - name: https://github.com/AlanCoding/awx.git#awx_collection,ee_req
+    type: git
+  - name: https://github.com/AlanCoding/azure.git
+    version: ee_req
+    type: git
+"""
+
+
+IMPORTED_REQUIREMENTS_YML = {
+    'collections': [
+        {
+            'name': 'https://github.com/AlanCoding/awx.git#awx_collection,ee_req',
+            'type': 'git'
+        },
+        {
+            'name': 'https://github.com/AlanCoding/azure.git',
+            'version': 'ee_req',
+            'type': 'git'
+        },
+    ]
+}
+
+
+REQUIREMENTS_TXT = """
+ansible
+paramiko
```

Comment:
these might not be great examples... because they are liable to be in the list of requirements we sanitize

https://github.com/ansible/ansible-builder/blob/devel/ansible_builder/requirements.py#L4

Whether or not you sanitize the list of requirements here or not is a very important and very big question. Since you're pulling in the python requirements parser library, that would indicate that you _are_ planning to do requirements sanitization, and that will duplicate more code from `ansible-builder` beyond the `introspect.py` script. I think you could get by without doing that sanitization. If you do pull that in, we will have two code bases with a lot of logic which will quickly start to diverge and it's going to get really hard to maintain.